tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -1294,6 +1294,106 @@ def assert_shape_match(inp, out):
     return (res_outputs, new_state)
 
 
+class LayerNormBasicGRUCell(rnn_cell_impl.RNNCell):
```

Comment:
Can you subclass `LayerRNNCell` instead, add a proper `build` method which calls `add_variable` and replace `__call__` with `call` that accesses variables via `self` properties?

Reply:
Hi @ebrevdo,
I have 8 calls to internal methods using 2 variables with different scope each time. I can't find an elegant way of using the `add_variable` and accessing variables with `self` properties while keeping code factorization. Can you provide me with such an example?

Reply:
Hello @ebrevdo,
Sorry for the delay, I did the refactoring, can you check that it was what you had in mind please?
Thanks

Reply:
This still doesn't subclass LayerRNNCell?